irb120: #namespace

    #--- Task Related parameters
    n_actions: 6 # The actions are the joint positions
    n_observations: 4 # The observations are XYZ positions of the TCP and the distance to GOAL
    max_steps: 10000

    #-----------------------------------------------------------------------------------------#
    #--- Action Space

    # Define min and max joint values to define 
    min_joint_pos:
      - -2.87979
      - -1.91986
      - -1.91986
      - -2.79253
      - -2.094395
      - -6.98132
    max_joint_pos:
      - 2.87979
      - 1.91986
      - 1.22173
      - 2.79253
      - 2.094395
      - 6.98132

    #-----------------------------------------------------------------------------------------#
    #--- Observation Space
    
    # Set
    position_ee_max: # Maximum XYZ position for the EE
      x: 1.0 
      y: 1.0
      z: 1.0
    position_ee_min: # Minimum XYZ position for the EE
      x: -1.0
      y: -1.0
      z: -1.0

    position_goal_max: # Maximum XYZ position for the goal
      x: 1.0
      y: 1.0
      z: 2.0
    position_goal_min: # Minimum XYZ position for the goal
      x: -1.0
      y: -1.0
      z: -1.0
      
    max_distance:     9.0 # Maximum distance from EE to the desired GOAL EE

    #-----------------------------------------------------------------------------------------#
    #--- Simulation parameters
    
    time_step: 0.01
    update_rate_multiplier: 20.0 # If training multiply the max update rate by this number 
    
    #-----------------------------------------------------------------------------------------#
    #--- Set goal EE pos
    
    goal_ee_pos: 
      x: 2.0
      y: 2.0
      z: 2.0

    tolerance_goal_pos: 0.1 # Tolerance for the position of the goal
    tolerance_goal_ori: 0.25 # Tolerance for the orientation of the goal in radians 0.5->30 degrees 0.25->15 degrees

    training: True
    pos_dynamic: False
    pos_dynamic_params:
      x_max_amplitude: 0.1
      y_max_amplitude: 0.1
      z_max_amplitude: 0.1
      x_max_freq: 0.01
      y_max_freq: 0.01
      z_max_freq: 0.01

    #-----------------------------------------------------------------------------------------#
    #--- Set variables related to rewards
    
    step_reward: -1.0 # 0.01
    multiplier_dist_reward: 100.0
    multiplier_ori_reward:  1.0
    reached_goal_reward:  100.0 # 10.0
    joint_limits_reward: -1000.0 # - 50.0